### DISCUSSION TITLE
Issue with Dependency Management in CI/CD Using Soldeer and Forge

### DISCUSSION THREAD

**Discussion by alvinyusuf** [foundry-rs/foundry#9306] [Help] [answered]
I'm encountering an issue with dependency management in my CI/CD pipeline while using Soldeer to manage dependencies in a Foundry project. The problem occurs when running forge install during the CI/CD process on GitHub Actions, where it throws an error:

`Updating dependencies in /home/runner/work/swap/swap/dependencies
error: pathspec '/home/runner/work/swap/swap/dependencies' did not match any file(s) known to git
Error: git submodule exited with code 1
Error: Process completed with exit code 1.`

Here's the relevant portion of my GitHub Actions config:

`name: CI

on:
  push:
  pull_request:
  workflow_dispatch:

env:
  FOUNDRY_PROFILE: ci

jobs:
  check:
    strategy:
      fail-fast: true

    name: Swap-Dex Project
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
        with:
          submodules: recursive

      - name: Install Foundry
        uses: foundry-rs/foundry-toolchain@v1
        with:
          version: nightly

      - name: Install dependencies
        run: |
          forge install

      - name: Show Forge version
        run: |
          forge --version

      - name: Run Forge fmt
        run: |
          forge fmt --check
        id: fmt

      - name: Run Forge build
        run: |
          forge build --sizes
        id: build

      - name: Run Forge tests
        run: |
          forge test -vvv
        id: test
`

It seems like the pipeline is unable to recognize or fetch the dependencies as expected when Soldeer is used.

Has anyone encountered this specific error, and if so, how did you resolve it?
Is there any additional setup required for Soldeer in a CI/CD environment that might be missing here?
Are there known compatibility issues with Soldeer and GitHub Actions that I should be aware of?
Any help would be greatly appreciated. Thank you!

**Comment by zerosnacks** [comment, marked as answer]
Hi @alvinyusuf, currently we require `forge soldeer install` to be ran when you interact with a Foundry project that uses Soldeer. This is incompatible with `forge install`. To fix this you should update you CI file with

```
  - name: Install dependencies
    run: |
      forge soldeer install
```

**Reply by alvinyusuf** [reply]
Thanks a lot for the answer, it was super helpful!
